File: sentry-samples/sentry-samples-spring-boot-opentelemetry/src/main/java/io/sentry/samples/spring/boot/PersonController.java
package io.sentry.samples.spring.boot;

import org.slf4j.Logger;
import org.slf4j.LoggerFactory;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.PathVariable;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

@RestController
@RequestMapping("/person/")
public class PersonController {
  private final PersonService personService;
  private static final Logger LOGGER = LoggerFactory.getLogger(PersonController.class);

  public PersonController(PersonService personService) {
    this.personService = personService;
  }

  @GetMapping("{id}")
  Person person(@PathVariable Long id) {
    LOGGER.info("Loading person with id={}", id);
    throw new IllegalArgumentException("Something went wrong [id=" + id + "]");
  }

  @PostMapping
  Person create(@RequestBody Person person) {
    return personService.create(person);
  }
}

File: sentry-samples/sentry-samples-spring-boot-jakarta-kotlin/src/main/kotlin/io/sentry/samples/spring/boot/PersonController.kt
package io.sentry.samples.spring.boot

import org.slf4j.Logger
import org.slf4j.LoggerFactory
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.PathVariable
import org.springframework.web.bind.annotation.PostMapping
import org.springframework.web.bind.annotation.RequestBody
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RestController

@RestController
@RequestMapping("/person/")
class PersonController(val personService: PersonService) {
    companion object {
        val LOGGER: Logger = LoggerFactory.getLogger(PersonController::class.java)
    }

    @GetMapping("{id}")
    fun person(@PathVariable id: Long): Person {
        LOGGER.info("Loading person with id={}", id)
        throw IllegalArgumentException("Something went wrong [id=$id]")
    }

    @PostMapping
    fun create(@RequestBody person: Person): Person {
        return personService.create(person)
    }
}

data class Person(val firstName: String, val lastName: String)
